Fix shared params with exactly_one_of

milgner opened the pull request:
Tentative fix for #1288 - not sure whether it is the right approach but the specs are green :wink:

I can't shake the feeling that there's some unnecessary complexity in combination with the changes from #1625 so I'm open to suggestions.

milgner commented:
Alrighty, waiting for Travis to do its thing and if everything is green there as well, I'll update the changelog...

rnubel commented:
It's been a while since I looked into this, but I like the simplicity of your fix. My own attempt at resolving the bug is https://github.com/ruby-grape/grape/pull/1691/files, but I don't like how it involved updating all of the validators individually (which is why I stalled on it). However, the specs that I added in that PR might be useful as additional test cases for this PR.

dblock commented on spec/integration/multi_xml/xml_spec.rb:
```diff
@@ -1,7 +1,13 @@
 require 'spec_helper'
 
 describe Grape::Xml do
```
This shouldn't be necessary, integration specs run in isolation. Is this failing on HEAD?

dblock commented:
Fix looks good to me. The XML adapter spec should be undone (run `rake` for tests) otherwise it's all good.

milgner commented:
Sorry, I was using `rspec` to run the tests and it was failing, so I fixed it. But using `rake`, it works as expected so I dropped the commit from the branch again.
